File: casa_guido/app/src/main/java/br/com/casa_guido/room/entidades/Pessoa.kt
package br.com.casa_guido.room.entidades

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(tableName = "pessoa")
data class Pessoa(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val nome: String = "",
    val contato: String = "",
    val dataNascimento: String,
    val cpf: String = "",
    val identidade: String = "",
    val naturalidade: String = "",
    val genero: String = "",
    val escolaridade: Int = 0,
    val serie: Int = 0,
    val estadoCivil: Int? = null,
    val situacaoProfissional: Int? = null,
    val salario: String = "",
    val enderecoId: String = "",
    val telefone: String = "",
    val cartaoSus: String = "",
    val profissao: String = "",
    val respPrincipal: Int = 0,
)

/*
data class Pessoa(
    val id: String = UUID.randomUUID().toString(),
    val nome: String = "",
    val telefone: String = "",
    val dataNascimento: String = "",
    val cpf: String = "",
    val rg: String = "",
    val endereco: Endereco = Endereco(),
    val situacaoProfissional : Int = 0,
    val estadoCivil : Int = 0,
    val naturalidade: String = "",
    val escolaridade: String = "",
    val salario: String = "",
)

 */

File: backend/models/Pessoa.js
const mongoose = require('mongoose');

const EnderecoSchema = new mongoose.Schema({
  cep: String,
  logradouro: String,
  numero: String,
  complemento: String,
  unidade: String,
  bairro: String,
  localidade: String,
  uf: String,
  estado: String,
  regiao: String,
  referencia: String,
  pais: { type: String, default: 'Brasil' }
}, { _id: false });

const PessoaSchema = new mongoose.Schema({
  nome: { type: String, required: true },
  telefone: String,
  dataNascimento: String,
  cpf: { type: String, unique: true, sparse: true },
  rg: { type: String, unique: true, sparse: true },
  naturalidade: String,
  genero: String,
  escolaridade: Number,
  serie: Number,
  estadoCivil: Number,
  situacaoProfissional: Number,
  salario: String,
  cartaoSus: String,
  respPrincipal: Number,
  profissao: String,
  endereco: EnderecoSchema
});

module.exports = mongoose.model('Pessoa', PessoaSchema); 